src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.opendistroforelasticsearch.knn.plugin.script;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+
+public class KNNScoringUtil {
+    private static Logger logger = LogManager.getLogger(KNNScoringUtil.class);
+
+    public static float l2Squared(float[] queryVector, float[] inputVector) {
+        float squaredDistance = 0;
+        for (int i = 0; i < inputVector.length; i++) {
+            float diff = queryVector[i]-inputVector[i];
+            squaredDistance += diff * diff;
+        }
+        return squaredDistance;
+    }
+
+    public static float cosinesimilOptimized(float[] queryVector, float[] inputVector, float normQueryVector) {
```

Comment:
Seems this method is mostly the same with the method below. Can we unify them by abstracting the common part?

Reply:
If we need to make some change in future, we may need to change both methods. But seems it's not high possible that we will change these two methods as they contain just simple calculation. I'm ok to keep it as is. 
How about we add some comments to these methods to explain differences and the best use cases ? I think it will help others to understand.